inductiva/benchmarks/methods.py

```diff
@@ -0,0 +1,351 @@
+""" 
+Methods to interact with Benchmarks.
+"""
+from collections import defaultdict
+import datetime
+import time
+import json
+
+from typing import Any, Callable, Dict, List, Union
+
+from inductiva.resources.machines_base import BaseMachineGroup
+from inductiva.templating.manager import TemplateManager
+from inductiva.resources.machines import MachineGroup
+from inductiva.simulators.simulator import Simulator
+from inductiva.projects.project import Project
+from inductiva.resources import machine_groups
+from inductiva.tasks.task import Task
+from inductiva import types, users
+
+from ..localization import translator as __
+
+
+def analyze(name: str,
+            metadata_path: types.Path,
+            ouput_path: types.Path = None) -> Dict:
+    """Analyzes a benchmark.
+    This method creates a dictionary with all the relevant information about
+    the benchmark.
+    Args:
+        name (str): Name of the benchmark.
+        metadata_path (types.Path): Path to the metadata file.
+        ouput_path (types.Path): Path to the output file.
+            If it will not create an output file.
+    Returns:
+        Dict: Dictionary with the information about the benchmark.
+    """
+
+    project = Project(name, append=True)
+    project.open()
+
+    tasks = project.get_tasks()
+
+    report = {"name": name, "resources": {}}
+
+    #reads the metadata file and creates a dictionary with the task_id as key
+    #and the metadata as value
+    metadata_tasks = {}
+    with open(metadata_path, "r", encoding="utf-8") as metadata_file:
+        lines = metadata_file.readlines()
+        for line in lines:
+            temp_task = json.loads(line)
+            metadata_tasks[temp_task["task_id"]] = temp_task
+
+    for task in tasks:
+
+        if task.get_status() != "success":
+            continue
+
+        task_info = task.get_info()
```

Comment:
After syncing the development branch this line should be deleted since the objects returned by `project.get_tasks()` already contain the task info available as an object (`task.info`).